driver/csiplugin/controllerserver.go

```diff
@@ -1644,6 +1680,58 @@ func (cs *ScaleControllerServer) ControllerPublishVolume(ctx context.Context, re
 	return &csi.ControllerPublishVolumeResponse{}, nil
 }
 
+func (cs *ScaleControllerServer) CheckNewSnapRequired(conn connectors.SpectrumScaleConnector, filesystemName string, filesetName string, snapWindow string) (string, error) {
+	latestSnapList, err := conn.GetLatestFilesetSnapshots(filesystemName, filesetName)
+	if err != nil {
+		glog.Errorf("CheckNewSnapRequired - getting latest snapshot list failed for fileset: [%s:%s]. Error: [%v]", filesystemName, filesetName, err)
+		return "", err
+	}
+
+	if len(latestSnapList) == 0 {
+		// No snapshot exists, so create new one
+		return "", nil
+	}
+
+	timestamp, err := cs.getSnapshotCreateTimestamp(conn, filesystemName, filesetName, latestSnapList[0].SnapshotName)
+	if err != nil {
+		glog.Errorf("error getting create timestamp for snapshot %s:%s:%s", filesystemName, filesetName, latestSnapList[0].SnapshotName)
+		return "", err
+	}
+
+	var timestampSecs int64
+	timestampSecs = timestamp.GetSeconds()
+	lastSnapTime := time.Unix(timestampSecs, 0)
+	passedTime := time.Now().Sub(lastSnapTime).Seconds()
+	glog.V(3).Infof("for fileset [%s:%s], last snapshot time: [%v], current time: [%v], passed time: %v seconds, snapWindow: %v minutes", filesystemName, filesetName, lastSnapTime, time.Now(), int64(passedTime), snapWindow)
+
+	snapWindowInt, err := strconv.Atoi(snapWindow)
+	if err != nil {
+		return "", status.Error(codes.Internal, fmt.Sprintf("CheckNewSnapRequired - invalid snapWindow value: [%v]", snapWindow))
+	}
+	snapWindowSeconds := snapWindowInt * 60
+
+	if passedTime < float64(snapWindowSeconds) {
+		// we don't need to take new snapshot
+		glog.V(3).Infof("CheckNewSnapRequired - for fileset [%s:%s], using existing snapshot [%s]", filesystemName, filesetName, latestSnapList[0].SnapshotName)
+		return latestSnapList[0].SnapshotName, nil
+	}
+
+	glog.V(3).Infof("CheckNewSnapRequired - for fileset [%s:%s] we need to create new snapshot", filesystemName, filesetName)
+	return "", nil
+}
+
+func (cs *ScaleControllerServer) MakeSnapMetadataDir(conn connectors.SpectrumScaleConnector, filesystemName string, filesetName string, indepFileset string) (error) {
+	path := fmt.Sprintf("%s/csi-snaphot-metadata/%s", indepFileset, filesetName) 
+	glog.V(3).Infof("MakeSnapMetadataDir - creating directory [%s] for fileset: [%s:%s]", path, filesystemName, filesetName)
+	err := conn.MakeDirectory(filesystemName, path, "0", "0")
```

Comment:
suppose user takes the 2 snapshot of same pvc and those are not within snapWindow then how this is taken care ?

Reply:
This has been taken care in last week's commit.